src/dso_api/dynamic_api/filters/parser.py

```diff
@@ -0,0 +1,379 @@
+"""The engine for filtering from the query-string.
+
+This translates a query-string into ORM lookups using the Amsterdam Schema definitions.
+"""
+from __future__ import annotations
+
+import operator
+import re
+from datetime import datetime
+from functools import reduce
+from typing import Any
+
+from django.core.exceptions import ValidationError as DjangoValidationError
+from django.db import models
+from django.db.models import Q
+from django.utils.datastructures import MultiValueDict
+from gisserver.geometries import CRS
+from rest_framework.exceptions import PermissionDenied, ValidationError
+from schematools.contrib.django.models import DynamicModel
+from schematools.exceptions import SchemaObjectNotFound
+from schematools.permissions import UserScopes
+from schematools.types import DatasetFieldSchema, DatasetTableSchema
+from schematools.utils import to_snake_case
+
+from .values import str2bool, str2geo, str2isodate, str2number, str2time
+
+# Allow notations: field.subfield[lookup]
+RE_KEY = re.compile(r"\A(?P<path>[\w\d_-]+(?:\.[\w\d_-]+)*)(?:\[(?P<lookup>[a-z0-9A-Z_-]+)])?\Z")
+
+# Lookups that have specific value types:
+LOOKUP_PARSERS = {
+    "isnull": str2bool,
+    "isempty": str2bool,
+    "like": str,  # e.g. uri is parsed as string for like.
+}
+
+MULTI_VALUE_LOOKUPS = {"not": operator.and_}
+
+SCALAR_PARSERS = {
+    "boolean": str2bool,
+    "string": lambda f: f,
+    "integer": int,
+    "number": str2number,
+}
+
+FORMAT_PARSERS = {
+    "string": {
+        "date": str2isodate,
+        "date-time": str2isodate,
+        "time": str2time,
+        "uri": str,
+    }
```

Comment:
Nitpick, but surely only `string` is ever going to have multiple formats, so we could hardcode that instead of looking it up in this dict.

Reply:
In OpenAPI, integer and number can also have a `format` argument. So this is more to be generic and exact.

Reply:
That sounds better. In JSON Schema, I don't think `number` can have a format.